### PR TITLE
cypress: improve tests

### PR DESCRIPTION
**Pull request by AoNoOokami**
* Removes useless code.
* Replaces UI actions by API calls where possible in order to save time
during the tests.
* Replaces UI navigation by cy.visit instruction when it's relevant
for the test.
* Adds a method to get current date and hour in order to use it in the
API requests.
* Adds methods to create a document and an item with API call.
* Updates fixtures for the requests body.

Co-Authored-by: Alicia Zangger <alicia.zangger@rero.ch>

## Why are you opening this PR?

To improve and fix cypress tests (they will need less time).

## Dependencies

My PR depends on the following `rero-ils-ui`'s PR(s):

* dev of rero-ils-ui
* dev of ng-core

## How to test?

You may need a bootstrap and a setup, depending on your local state.
You must have the correct branchs of rero-ils-ui and ng-core locally (see dependencies above).
`poetry run folding -c path/to/rero/ng-core -u path/to/rero/rero-ils-ui`
`FLASK_DEBUG=False poetry run server`
`poetry run cypress`

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?

### CODE REVIEW

**Review by zannkukai — APPROVED** [review]

File: tests/e2e/cypress/cypress/integration/examples/example-test-request.spec.js [hunk 1]
```diff
@@ -81,46 +83,52 @@ describe('Template: librarian request', function() {
...
   });
 
   // First test
   it('First test: a librarian makes a request for a patron', function() {
     console.log('first test');
-    cy.route('/api/item/*/can_request?library_pid=' + this.users.librarians.spock.libraryPid + '&patron_barcode=' + this.users.patrons.james.barcode).as('getCanRequest');
+    cy.route('/api/item/*/can_request?library_pid='
+      + this.users.librarians.spock.libraryPid
+      + '&patron_barcode='
+      + this.users.patrons.james.barcode)
```
Comment (Garfield-fr) [inline tests/e2e/cypress/cypress/integration/examples/example-test-request.spec.js:106]: cy.route(\`/api/item/*/can_request?
library_pid=${this.users.librarians.spock.libraryPid}
&patron_barcode=${this.users.patrons.james.barcode}\`)

File: tests/e2e/cypress/cypress/integration/templates/templates-test.spec.js [hunk 2]
```diff
@@ -27,7 +27,7 @@ before(function () {
   });
   cy.fixture('templates').then(function (templateData) {
     this.templates = templateData;
-    this.templateName = this.templates.templateA.name + cy.getCurrentDateAndHour();
+    this.templateName = this.templates.templateA.name + ' ' + cy.getCurrentDateAndHour();
```
Comment (Garfield-fr) [inline tests/e2e/cypress/cypress/integration/templates/templates-test.spec.js:30]: `[this.templates.templateA.name, cy.getCurrentDateAndHour()].join(' ')`

File: tests/e2e/cypress/cypress/support/api.js [hunk 3]
```diff
@@ -17,43 +17,102 @@ along with this program. If not, see <http://www.gnu.org/licenses/>.
...
+  */
+ Cypress.Commands.add('apiCreateItem', (item, barcode, documentPid) => {
+  cy.request({
+    method: 'POST',
+    url: '/api/items/',
+    followRedirect: false,
+    body: {
+      "acquisition_date":cy.getCurrentDate(),
+      "item_type":{
+        "$ref":('https://ils.rero.ch/api/item_types/' + item.itemTypePid)
```
Comment (Garfield-fr) [inline tests/e2e/cypress/cypress/support/api.js:93]: "$ref":\`https://ils.rero.ch/api/item_types/${item.itemTypePid}`

File: tests/e2e/cypress/cypress/support/api.js [hunk 4]
```diff
@@ -17,43 +17,102 @@ along with this program. If not, see <http://www.gnu.org/licenses/>.
...
-    }).then(response => {
-      cy.visit('/')
-      cy.get('#logout-menu')  // raise an error if not found
-    })
+      ,"type":"standard",
+      "status":"on_shelf"
+      ,"barcode":barcode
+      ,"call_number":barcode
+      ,"document":{
+        "$ref":('https://ils.rero.ch/api/documents/' + documentPid)
```
Comment (Garfield-fr) [inline tests/e2e/cypress/cypress/support/api.js:103]: same here

File: tests/e2e/cypress/cypress/support/api.js [hunk 5]
```diff
@@ -17,43 +17,102 @@ along with this program. If not, see <http://www.gnu.org/licenses/>.
...
+    method: 'POST',
+    url: '/api/items/',
+    followRedirect: false,
+    body: {
+      "acquisition_date":cy.getCurrentDate(),
+      "item_type":{
+        "$ref":('https://ils.rero.ch/api/item_types/' + item.itemTypePid)
+      },
+      "location":{
+        "$ref":('https://ils.rero.ch/api/locations/' + item.locationPid)
```
Comment (Garfield-fr) [inline tests/e2e/cypress/cypress/support/api.js:96]: same hier

File: tests/e2e/cypress/cypress/support/api.js [hunk 6]
```diff
@@ -17,43 +17,102 @@ along with this program. If not, see <http://www.gnu.org/licenses/>.
...
+    }
   })
-})
-
+  .its('body').then((body) => {
+    cy.wrap(body.id).as('getItemPid');
+  })
+  .then(() => {
+    cy.get('@getItemPid').then((pid) => {
+      cy.log('Item created, pid = ' + pid);
```
Comment (Garfield-fr) [inline tests/e2e/cypress/cypress/support/api.js:112]: \`Item created, pid = ${pid}\`

File: tests/e2e/cypress/cypress/support/api.js [hunk 7]
```diff
@@ -67,9 +126,9 @@ Cypress.Commands.add('apiDeleteResources', (resourceName, query) => {
       const pid = hit.metadata.pid
       cy.request({
         method: 'DELETE',
-        url: '/api/'+resourceName+'/'+pid
+        url: '/api/' + resourceName+'/'+ pid
```
Comment (Garfield-fr) [inline tests/e2e/cypress/cypress/support/api.js:129]: `['api', resourceName, pid].join('/')]`

File: tests/e2e/cypress/cypress/support/api.js [hunk 8]
```diff
@@ -67,9 +126,9 @@ Cypress.Commands.add('apiDeleteResources', (resourceName, query) => {
...
       })
-      cy.log('Template#'+pid+' deleted')
+      cy.log(resourceName + ' #' + pid + ' deleted')
```
Comment (Garfield-fr) [inline tests/e2e/cypress/cypress/support/api.js:131]: same here

File: tests/e2e/cypress/cypress/support/navigation.js [hunk 9]
```diff
@@ -29,25 +29,23 @@ Cypress.Commands.add("userProfile", (tabId) => {
...
-  cy.get('.form-control').type(itemBarcode).type('{enter}');
-  // Use first element
-  cy.wait('@getDocumentPermissions');
-  cy.get('[name=document-title]').click();
-})
+Cypress.Commands.add("goToPublicDocumentDetailView", (pid, viewcode) => {
+  if (viewcode == undefined) {
+    cy.visit('/global/documents/' + pid);
+  } else {
+    cy.visit('/' + viewcode + '/documents/' + pid);
```
Comment (Garfield-fr) [inline tests/e2e/cypress/cypress/support/navigation.js:36]: same here

File: tests/e2e/cypress/cypress/support/navigation.js [hunk 10]
```diff
@@ -29,25 +29,23 @@ Cypress.Commands.add("userProfile", (tabId) => {
...
-  // on professional context
-  cy.get('.form-control').clear();
-  cy.get('.form-control').type(itemBarcode).type('{enter}');
-  // Use first element
-  cy.wait('@getDocumentPermissions');
-  cy.get('[name=document-title]').click();
-})
+Cypress.Commands.add("goToPublicDocumentDetailView", (pid, viewcode) => {
+  if (viewcode == undefined) {
+    cy.visit('/global/documents/' + pid);
```
Comment (Garfield-fr) [inline tests/e2e/cypress/cypress/support/navigation.js:34]: same here

File: tests/e2e/cypress/cypress/support/navigation.js [hunk 11]
```diff
@@ -29,25 +29,23 @@ Cypress.Commands.add("userProfile", (tabId) => {
...
+  } else {
+    cy.visit('/' + viewcode + '/documents/' + pid);
+  }
+});
+
+Cypress.Commands.add("goToProfessionalDocumentDetailView", (pid) => {
+  cy.visit('/professional/records/documents/detail/' + pid);
```
Comment (Garfield-fr) [inline tests/e2e/cypress/cypress/support/navigation.js:41]: same here

File: tests/e2e/cypress/cypress/support/navigation.js [hunk 12]
```diff
@@ -29,25 +29,23 @@ Cypress.Commands.add("userProfile", (tabId) => {
...
+});
+
+/**
+ * Go to item detailed view. Needs to have the item pid stored in '@getItemPid' alias
+ *
+ * @param itemPid - item pid
+ */
+Cypress.Commands.add("goToItemDetailView", (pid) => {
+    cy.visit('/professional/records/items/detail/' + pid);
```
Comment (Garfield-fr) [inline tests/e2e/cypress/cypress/support/navigation.js:50]: same here

**Review by Garfield-fr — APPROVED** [review]

File: tests/e2e/cypress/cypress/integration/templates/templates-test.spec.js (hunk 2 above)
Comment (AoNoOokami) [reply tests/e2e/cypress/cypress/integration/templates/templates-test.spec.js:30]: Done

File: tests/e2e/cypress/cypress/support/api.js (hunk 7 above)
Comment (AoNoOokami) [reply tests/e2e/cypress/cypress/support/api.js:129]: Done

File: tests/e2e/cypress/cypress/support/api.js (hunk 8 above)
Comment (AoNoOokami) [reply tests/e2e/cypress/cypress/support/api.js:131]: Done

**Pull request merged**